### PR TITLE
Update dependencies to support react 17

### LINKED ISSUE #8283: Investigate updating @woocommerce/components to support React 17
It was recently pointed out to us that `@woocommerce/components` is causing difficulties for consumers due to the current React 16 dependency. (pca54o-2Bu-p2)

There is a web of dependencies to untangle to allow an update to React 17. Some research was undertaken in this “[Renovate Bot” PR](https://github.com/woocommerce/woocommerce-admin/pull/7510).

This investigation should investigate updating necessary dependencies to support React 17.

### PR DESCRIPTION
**Pull request by chihsuan**
Fixes #8283

This PR updates the dependencies to support react 17 with the following changes.

- Updated dependencies to support react 17. d33b4ea91a888857d74e46bba551f62594907b1a
- Updated test snapshots due to wp component package updated. 734749d8cd59e02d9e353be39e3894d0e3c955ad
- ~~Added `explat-client-react-helpers.tsx` from `wp-calypso/packages/explat-client-react-helpers` to `pacakges/explat` as temporary solution to support react 17. https://github.com/woocommerce/woocommerce-admin/pull/8305/commits/22befc90c24f916e524e6c5153e8bcb21bfce718~~ -> Update webpack.config for latest published explat-client-react-helpers. c2fc69a33094dde39a03aa1f99ae434b63327da3
- Updated `SnackbarList` with latest react-spring. 37ca4a505947509dd727623c20db3330b5bd5d9c
- Removed `'@wordpress/components/src/visually-hidden/style'` import (included in wp-components). 42a4dde3235a6d0601c93ae4309b890341a8fdee
- Updated `interpolateComponents` import path -> `'@automattic/interpolate-components'` 18b60b9e03c1a53cdc2a6f668c5fbc50a04fa53c
- ~~Updated `createNotice` actions param `onclick` (deprecated) -> `callback`. 4666b8f0ca5300665db08fbe4d2ccdfe8fc6f0a9~~ -> https://github.com/woocommerce/woocommerce-admin/pull/8305#pullrequestreview-885490195
- Fixed task-list-item and display-options tests. a00c320e44960f20c4ee5c6aaa22be876b303625 2b03298d7a2cc39f43918ea3d99e90ab6965b271

Details: p90Yrv-2Lk-p2

### Detailed test instructions:

- Check out this branch
- Clean repo: `npm run clean && rm -rf node_modules && rm -rf packages/**/node_modules`
- Reinstall and build: `npm install && npm run build`
- Test: `npm run test`
- Smoke test the whole app

### CODE REVIEW

**Review by rjchow — APPROVED** [review]
Nice work! Appreciate this very laborious and often thankless work.
Automated tests and smoke testing went well, almost all the changes look fine.

I went over the changelogs of every package that was updated and the only things that stood out were:
- Most of them had breaking changes of IE11 deprecation and Node.JS LTS 12 as the minimum version - wonder if we should add this to our own changelog?
- @wordpress/components deprecated some props (isPrimary, isSecondary, isTertiary, isLink) in `<Button>` and we use those props, so we should probably make a separate issue to fix them before they completely remove support. https://github.com/WordPress/gutenberg/blob/81e8e0f7c999df3cc43ef97cb2913779c0190eff/packages/components/CHANGELOG.md#deprecation-1

**Comment by chihsuan** [comment]
>     * Most of them had breaking changes of IE11 deprecation and Node.JS LTS 12 as the minimum version - wonder if we should add this to our own changelog?
> 
>     * @wordpress/components deprecated some props (isPrimary, isSecondary, isTertiary, isLink) in `<Button>` and we use those props, so we should probably make a separate issue to fix them before they completely remove support. https://github.com/WordPress/gutenberg/blob/81e8e0f7c999df3cc43ef97cb2913779c0190eff/packages/components/CHANGELOG.md#deprecation-1

@rjchow Thanks for spotting them. I've add IE11 deprecation note to changelogs in  bd4a659f791e3a618bdca99ce3b8bb8db4f0690e. I think our minimum Node JS version is already 12 as defined in the package.json.

Also created a follow-up issue https://github.com/woocommerce/woocommerce-admin/issues/8313 for `<Button>`

**Review by ilyasfoo — CHANGES_REQUESTED** [review]
Thanks so much for the work here, @chihsuan! It's super neat that you've also properly separated the changes into different commits, definitely helped with reviewing there.

I've found an issue with `createNotice` changes. Everything else is looking good so far!

File: client/homescreen/activity-panel/reviews/index.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ class ReviewsPanel extends Component {
 							actions: [
 								{
 									label: __( 'Undo', 'woocommerce-admin' ),
-									onClick: () => {
+									callback: () => {
```
[the remark this answers is not in the document]
Comment (chihsuan) [reply client/homescreen/activity-panel/reviews/index.js:72]: Good catch. I think you're right. It's still used `onClick`.  I found the `callback` from typescript warnings and `@types/wordpress__notices`.


Screen Shot 2022-02-17 at 16 45 44
Screen Shot 2022-02-17 at 16 45 53
.

[the remark this answers is not in the document]
Comment (chihsuan) [reply client/homescreen/activity-panel/reviews/index.js:72]: https://www.npmjs.com/package/@types/wordpress__notices

The last published is a year ago. : (

[the remark this answers is not in the document]
Comment (chihsuan) [reply client/homescreen/activity-panel/reviews/index.js:72]: Let me drop this commit https://github.com/woocommerce/woocommerce-admin/commit/4666b8f0ca5300665db08fbe4d2ccdfe8fc6f0a9 and possibly fill an issue on Gutenberg.

[the remark this answers is not in the document]
Comment (chihsuan) [reply client/homescreen/activity-panel/reviews/index.js]: I've reverted the changes and confirmed they actually changed from `callback` to `onClick` (https://github.com/WordPress/gutenberg/pull/17206), but [didn't update the file in DefinitelyTyped repo](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__notices/index.d.ts#L32).

[the remark this answers is not in the document]
Comment (ilyasfoo) [reply client/homescreen/activity-panel/reviews/index.js]: Ah, I see. Yeah it seems that'll need an update

[the remark this answers is not in the document]
Comment (chihsuan) [reply client/homescreen/activity-panel/reviews/index.js]: Update: I've raised a discussion https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/58850 based on https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377.

**Review by ilyasfoo — APPROVED** [review]
I confirm that `createNotice` is working again. LGTM!

**Review by adrianduffell — APPROVED** [review]
Smoke test went well for me! Nice job on this @chihsuan 👍

File: client/homescreen/activity-panel/reviews/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (chihsuan) [reply client/homescreen/activity-panel/reviews/index.js]: I tagged the other maintainers from `@types/wordpress__components` 🥲

**Pull request merged**